core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala

```diff
@@ -119,18 +119,23 @@ private[spark] object ResourceUtils extends Logging {
   def parseAllResourceRequests(
       sparkConf: SparkConf,
       componentName: String): Seq[ResourceRequest] = {
-    listResourceIds(sparkConf, componentName).map { id =>
-      parseResourceRequest(sparkConf, id)
+    listResourceIds(sparkConf, componentName).flatMap { id =>
+      val req = parseResourceRequest(sparkConf, id)
+      if (req.amount > 0) Some(req) else None
     }
```

Comment:
Can we use `.filter` here?

Reply:
we could but we would have to break it apart into 2 steps - one that gets the parsedResourceRequests back then one that filters that list of requests because we don't have the amount until we parseResourceRequest. I'm not sure it matter from performance point?

Reply:
Ya. It's just a style issue. We can ignore that.

Reply:
Yeah we are saying the same thing.  the .map().filter() may be easier to read. I'll update it.